heron/statefulstorages/src/java/com/twitter/heron/statefulstorage/localfs/LocalFSStorage.java

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.statefulstorage.localfs;
+
+import java.io.File;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import com.twitter.heron.common.basics.FileUtils;
+import com.twitter.heron.proto.ckptmgr.CheckpointManager;
+import com.twitter.heron.spi.statefulstorage.Checkpoint;
+import com.twitter.heron.spi.statefulstorage.IStatefulStorage;
+
+public class LocalFSStorage implements IStatefulStorage {
```

Comment:
Local and hdfs storage are basically the same with the exception of how files are read/written. Can we please use a common abstract superclass for consistency and less code duplication? Similar to what we did with `FileSystemStateManager`. I'm already looking at this class and needing to make the same comments twice due to copied code.